src/uct/ib/base/ib_iface.h

```diff
@@ -535,8 +536,9 @@ void uct_ib_iface_fill_ah_attr_from_gid_lid(uct_ib_iface_t *iface, uint16_t lid,
     ah_attr->port_num          = iface->config.port_num;
     ah_attr->grh.traffic_class = iface->config.traffic_class;
 
-    if (iface->is_global_addr ||
-        (iface->gid.global.subnet_prefix != gid->global.subnet_prefix)) {
+    if (iface->config.is_global || /* forced global address by user */
```

Comment:
if this forced by user, then it has to be applied there:
https://github.com/openucx/ucx/blob/92bba4a937b4a027b4d482fa469b24f0434786b3/src/uct/ib/base/ib_iface.c#L892
and if I'm not mistaken this check (and parameter in iface's config) is excessive and can be removed

Reply:
there are 2 locations where global addressing is configured: in iface & in endpoint. iface is responsible to pack global/local address, EP - which part to use (global/local). in case if user forced global addressing - we are always using it, if user has no such ideas - then we make decision about global addressing based on network configuration + different subnet

Reply:
@hoopoepg thanks, I see now. it just to not break a case when a user sets this env to use global addressing (but `iface->gid.global.subnet_prefix` == `gid->global.subnet_prefix`)

Reply:
but why need another bool var on iface? If user forced global addr, self->is_global_addr will be instantiated accordingly

Reply:
another var is used as "cache" for local/global addr tp pack global/local address

Reply:
looks like previously it was impossible to change behavior, right? since, other parameters can make `iface->is_global_addr=1`
now using another var we can force using global address
btw, do we really need to support case when it's global and we force it as local? I guess it shouldn't work

Reply:
it may force "global" only, not "local"

Reply:
> it may force "global" only, not "local"

yes, right
looks like then we don't need a separate bool variable, since it can adjust `iface->is_global_addr` and we will use it here in the following way:
```
if (iface->is_global_addr ||
    (iface->is_global_addr &&
     (iface->gid.global.subnet_prefix != gid->global.subnet_prefix))) {
    ...
}
```